pkg/systemd/generate/containers.go

```diff
@@ -236,7 +236,14 @@ func containerServiceName(ctr *libpod.Container, options entities.GenerateSystem
 	if options.Name {
 		nameOrID = ctr.Name()
 	}
-	serviceName := fmt.Sprintf("%s%s%s", options.ContainerPrefix, options.Separator, nameOrID)
+
+	var serviceName string
+	if options.ContainerPrefix == "" {
+		serviceName = nameOrID
```

Comment:
would you consider `serviceName := NameOrID` and then use a single IF condition for options.ContainerPrefix != "".  Also, personal thing, but i prefer len(options.ContainerPrefix) == 0 over "".

Reply:
I think it would be better to put this into a small function and use this function for container and pod name,
e.g.
```
function getServiceName(prefix, sperator, name) string {
   if prefix == "" {
      return name
   }
   return fmt.Sprintf("%s%s%s", prefix, separator, name)
}

serviceName := getServiceName(...)
```
This way would do not need to duplicate the logic

Reply:
I have created a function for this. I also used + to concatenate string instead of Sprintf because fmt was not imported in common.go and I wasn't sure if I should import it only for this function. I also changed the code to use a single if.